backend/src/main.py

```
@@ -81,12 +92,31 @@ def internal_error(error):
 
+if __name__ == '__main__':
+    # Initialize database
+    init_db()
+
+    # Start background asyncio services in a dedicated thread
+    bg_thread = threading.Thread(target=start_async_loop, daemon=True)
+    bg_thread.start()
+
+    # Run the Flask-SocketIO app
+    logger.info("Starting Flask-SocketIO server on port 5000...")
+    socketio.run(app, host='0.0.0.0', port=5000)
```

Ruff (0.15.6)

[error] Possible binding to all interfaces

(S104)